### PR TITLE
feat: resolver [STU-99]

### PR DESCRIPTION
**Pull request by billiegoose**
what we talked about on Slack. Only I wasted many hours trying to convert the `<DraftValue>` and `<ThrottleValue>` HOCs to Hooks thinking my memoizing was broken, but really formtron was just broken all along? checkboxes, IDK, will fix everything someday 😢

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -3,6 +3,8 @@ import * as React from 'react';
 import { Dictionary } from 'ts-essentials';
 import { themeTypes } from './theme';
 
+export type Resolver = (path: string[]) => any;
```
Comment (P0lip) [inline src/types.ts]:
```suggestion
export type Resolver = (path: JsonPath) => unknown;
```

File: src/deriveFormData.ts [hunk 2]
```diff
@@ -52,6 +53,9 @@ export const deriveFormData = memoize((schema: any, data: any, selection: string
         } else {
           // There's one edge case where path = ""
           output[key] = path === '' ? data : get(data, path);
+          if (output[key] === undefined && resolver) {
+            output[key] = resolver(path.split('.'));
```
Comment (P0lip) [inline src/deriveFormData.ts:57]: Seems like `resolver` should accept an array of numbers and/or strings.

Comment (billiegoose) [reply src/deriveFormData.ts:57]: all it ever gets is the output of `.split('.')` so....... `resolver` doesn't have to support numbers. I think if your resolver _did_ support numbers that wouldn't be a type error right? E.g. if

```ts
const myResolver: (path: Array<number | string>) => any
```

that would still work right?

File: src/types.ts (hunk 1 above)
Comment (billiegoose) [reply src/types.ts]: :+1: for the `unknown`. Not sure what the `JsonPath` refers to.

Edit: oooooooooooh

File: src/deriveFormData.ts (hunk 2 above)
Comment (billiegoose) [reply src/deriveFormData.ts:57]: ayayayaya nvm it all makes sense now

**Review by P0lip — APPROVED** [review]

**Pull request merged**